### PR TITLE
WIP PR: FlowManagerExtension struct

### PR DESCRIPTION
**Pull request by prmehra**
This is a WIP PR (still need to optimize, test & improve) but just published for collaboration purposes. A lot of the code comes from stuff @JacopoMangiavacchi had already written in his flow manager tool (thanks Jac!). 

I am now writing some client side code on the watch so that I can test this (simply creating a file on the watch and passing in the URL as a parameter in the FlowManager struct)

Addressed Jac's comments and is no longer a WIP PR but an actual one :) 
cc @Roberdan @JacopoMangiavacchi

### CODE REVIEW

File: Sources/FlowManager/FlowManager.swift [hunk 1]
```diff
@@ -4,6 +4,101 @@
 //  Part of the  EpilepsyResearchKit2020 Project
 
 import Foundation
+import CSV
+
+public struct FlowManagerExtension {
```
[the remark this answers is not in the document]
Comment (JacopoMangiavacchi) [reply Sources/FlowManager/FlowManager.swift:9]: I would recommend also to have the FlowManagerExtension protocol defined in the FlowManager.swift file but have the  FlowManagerCSVStorage struct implemented on a separated swift file.

File: Sources/FlowManager/FlowManager.swift [hunk 2]
```diff
@@ -502,6 +599,10 @@ public struct FlowManager {
         // Circular Buffers
         self.hrvRingAverage = RingAverage(size: parameters.hrvBufferSize, averageFunc: RingAverage.ssdnAverageFunction)
         self.speedRingAverage = RingAverage(size: parameters.averageSpeedBufferSize)
+        
+        // Storage
+        if (fileUrl != "") {
+            self.storageDelegate = FlowManagerExtension.init(url: fileUrl) }
```
Comment (JacopoMangiavacchi) [inline Sources/FlowManager/FlowManager.swift:605]: just something like ```self.delegate = delegate```

**Comment by prmehra** [comment]
Sounds good! Added a task to add unit tests

**Comment by prmehra** [comment]
Thank you Jac!!

**Comment by Roberdan** [comment]
hey @JacopoMangiavacchi  @prmehra since you merged (:-)), is time to create also a new release and update the package in the MVP app? thx rob

**Comment by JacopoMangiavacchi** [comment]
@Roberdan yes, I'm releasing the new tag right now.  The beauty of this PR is that the external interface didn't change as the 'storage extension' is an optional parameter of the constructor.  So super easy for a quick integration test on the MVP 1.0 App.  @prmehra can you test the new (coming) 0.1.17 tag with the watchOS master branch ?

**Comment by prmehra** [comment]
Yep! I’ll do it today

Thanks guys,
Pritika

**Pull request merged**